src/components/dar_collection_table/DarCollectionTable.js

```diff
@@ -103,28 +103,67 @@ const CancelCollectionButton = (props) => {
   });
 };
 
+const ResubmitCollectionButton = (props) => {
+  const { collection } = props;
+  return h(SimpleButton, {
+    keyProp: `resubmit-collection-${collection.id}`,
+    label: 'Revise',
+    //todo: should we filter out any statuses that could be resubmitted? is this the correct place for such logic?
+    //disabled: includes(determineCollectionStatus(collection))(['Under Election']),
+    baseColor: Theme.palette.secondary,
+    additionalStyle: {
+      width: '30%',
+      padding: '2%',
+      fontSize: '1.45rem',
+    },
+    onClick: () => props.showConfirmationModal(collection)
+  });
+};
+
 const actionsCellData = ({collection, showConfirmationModal}) => {
   const { darCollectionId } = collection;
   const cancelButtonTemplate = h(CancelCollectionButton, {
     collection,
     showConfirmationModal
   });
-
-  return {
-    isComponent: true,
-    id: darCollectionId,
-    label: 'cancel-button',
-    data: div(
-      {
-        style: {
-          display: 'flex',
-          justifyContent: 'left'
+  const resubmitButtonTemplate = h(ResubmitCollectionButton, {
+    collection,
+    showConfirmationModal
+  });
+  if(isCollectionCanceled(collection) === true)
+  {
+    return {
+      isComponent: true,
+      id: darCollectionId,
+      label: 'resubmit-button',
+      data: div(
+        {
+          style: {
+            display: 'flex',
+            justifyContent: 'left'
+          },
+          key: `resubmit-collection-cell-${darCollectionId}`
         },
-        key: `cancel-collection-cell-${darCollectionId}`
-      },
-      [cancelButtonTemplate]
-    )
-  };
+        [resubmitButtonTemplate]
+      )
+    };
+  } else {
+    return {
+      isComponent: true,
+      id: darCollectionId,
+      label: 'cancel-button',
+      data: div(
+        {
+          style: {
+            display: 'flex',
+            justifyContent: 'left'
+          },
+          key: `cancel-collection-cell-${darCollectionId}`
+        },
+        [cancelButtonTemplate]
+      )
+    };
```

Comment:
We have two returns in an if-else block that makes this a little difficult to read. Instead of embedding the object construction inside the if-else, it would be easier to reason about if that were defined outside. For example, we could have two objects defined that could be returned, the cancel object and the revise object, and then simplify the return, something like this:

```
  const cancel = {
    isComponent: true,
    id: darCollectionId,
    label: 'cancel-button',
    data: div(
      {
        style: {
          display: 'flex',
          justifyContent: 'left'
        },
        key: `cancel-collection-cell-${darCollectionId}`
      },
      [h(CancelCollectionButton, {collection, showConfirmationModal})]
    )};

  const revise = {
    isComponent: true,
    id: darCollectionId,
    label: 'resubmit-button',
    data: div(
      {
        style: {
          display: 'flex',
          justifyContent: 'left'
        },
        key: `resubmit-collection-cell-${darCollectionId}`
      },
      [h(ResubmitCollectionButton, {collection, showConfirmationModal})]
    )
  };

  return isCollectionCanceled(collection) ? revise : cancel;
```